### PR TITLE
batch code with thermostat

### PR DESCRIPTION
**Pull request by chuantian**

### CODE REVIEW

File: fe/parameterize_batch_thermostat.py [hunk 1]
```diff
@@ -0,0 +1,440 @@
...
+        ccs,
+        complete_lambda,
+        lambda_idxs,
+        precision
+    )
+
+    initial_params = sim.system.params
+
+    lr = 1e-4
+    opt_init, opt_update, get_params = optimizers.adam(lr)
```
Comment (proteneer) [inline fe/parameterize_batch_thermostat.py:233]: lr=1e-4 is probably too small for adam

Comment (proteneer) [reply fe/parameterize_batch_thermostat.py:233]: you can test this with a fixed seed and seed if adam can learn anything

Comment (chuantian) [reply fe/parameterize_batch_thermostat.py:233]: Got it!

File: fe/parameterize_batch_thermostat.py [hunk 2]
```diff
@@ -0,0 +1,440 @@
...
+    vary_allowed_groups = [(14, 1e-2)]*30 + [(12, 1e-2)]*30 + [(13, 1e-2)]*30
+
+    assert len(vary_allowed_groups) == num_epochs
+
+    ligand_list = []     
+
+    for ligand_sdf_file in args.ligand_sdf:
+        for rd_mol in Chem.SDMolSupplier(ligand_sdf_file, removeHs=False):
+            dG_value = rd_mol.GetProp("_dG")
+            mol_name = rd_mol.GetProp("s_user_Name")
```
Comment (proteneer) [inline fe/parameterize_batch_thermostat.py:249]: Something generally useful: if they don't encode the s_user_Name explicitly, you can use
```
rd_mol.GetProp('_Name')
```
To get the name (CHEMBL3402753_200) in the HEADER block off the sdf file:
```
CHEMBL3402753_200
                    3D
 Structure written by MMmdl.
 46 49  0  0  1  0            999 V2000
   24.6440   33.7566   50.5299 C   0  0  0  0  0  0
   24.1955   33.6079   51.9712 C   0  0  0  0  0  0
   22.9269   33.1279   52.2874 C   0  0  0  0  0  0
   22.5559   32.9819   53.6250 C   0  0  0  0  0  0
   23.4682   33.3170   54.6402 C   0  0  0  0  0  0
   24.7033   33.8055   54.3290 N   0  0  0  0  0  0
   25.0501   33.9349   53.0237 C   0  0  0  0  0  0
   23.1953   33.0558   56.0977 C   0  0  0  0  0  0
   24.2278   33.1076   57.0532 C   0  0  0  0  0  0
   24.0151   32.7150   58.3740 C   0  0  0  0  0  0
   22.7600   32.2678   58.7755 C   0  0  0  0  0  0
```

Comment (proteneer) [reply fe/parameterize_batch_thermostat.py:249]: In the hif2a ligands.sdf file they don't include the 's_user_Name' and call it the ID instead:

```
20 22  2  0  0  0
 20 27  1  0  0  0
 27 36  1  0  0  0
 27 37  1  0  0  0
M  END
> <ID>
338
```

Comment (chuantian) [reply fe/parameterize_batch_thermostat.py:249]: Yeah, that's why I use the shared "true_dG" I included as metric before, but if "_Name" works for all cases, I don't need the "ID" or "s_user_Name" or else.

Comment (proteneer) [reply fe/parameterize_batch_thermostat.py:249]: _Name should be always defined for a properly written SDF. @bp-kelley ?

File: fe/parameterize_batch_thermostat.py [hunk 3]
```diff
@@ -0,0 +1,440 @@
...
+                        if gp in allowed_groups:
+                            pf = allowed_groups[gp]
+                            filtered_grad.append(g*pf)
+                            if g != 0:
+                                print("derivs", g_idx, '\t group', gp, '\t', g, '\t adjusted to', g*pf, '\t old val', sim.system.params[g_idx])
+                        else:
+                            filtered_grad.append(0)
+
+                    filtered_grad = np.array(filtered_grad)
+                    opt_state = opt_update(epoch, filtered_grad, opt_state)
```
Comment (proteneer) [inline fe/parameterize_batch_thermostat.py:433]: needs to be next(itercount)

Comment (chuantian) [reply fe/parameterize_batch_thermostat.py:433]: Fixed!

**Comment by proteneer** [comment]
@chuan this is stale - so i'm closing it
